prov/rxd/src/rxd_cq.c

```diff
@@ -425,20 +443,29 @@ static void rxd_handle_rts(struct rxd_ep *ep, struct rxd_pkt_entry *pkt_entry)
 			"ERROR: Protocol version mismatch with peer\n");
 		return;
 	}
+    rxd_av = rxd_ep_av(ep);
+    fastlock_acquire(&rxd_av->util_av.lock);
 
-	rxd_av = rxd_ep_av(ep);
-	node = ofi_rbmap_find(&rxd_av->rbmap, pkt->source);
+    ret = rxd_av_insert_dg_addr(rxd_av, pkt->source, 
+                &dg_addr, 0, NULL);
+    if (ret){
+        fastlock_release(&rxd_av->util_av.lock);
+        return;
+    }
 
-	if (node) {
-		rxd_addr = (fi_addr_t) node->data;
-	} else {
-		ret = rxd_av_insert_dg_addr(rxd_av, (void *) pkt->source,
-					    &rxd_addr, 0, NULL);
-		if (ret)
-			return;
-	}
+    peer_entry = rxd_get_peer_entry(ep, pkt->source);
```

Comment:
Again - not sure where this is defined. I'm guessing this is part of the stuff you're working on. I understand now the addr/name being passed in but let's rename it to make it clear that we're looking up by name especially because at this point we're not guaranteed that the peer entry exists so it's more of a lookup than a get. If it's not there then we need to do some sort of initialization of the peer.

Reply:
rxd_ep.c